hibernate-core/src/main/java/org/hibernate/sql/ast/spi/AbstractSqlAstTranslator.java

```diff
@@ -6314,7 +6314,7 @@ protected void renderTableReferenceIdentificationVariable(TableReference tableRe
 	}
 
 	protected void registerAffectedTable(NamedTableReference tableReference) {
-		registerAffectedTable( tableReference.getTableExpression() );
+		registerAffectedTable( tableReference.getTableName() );
```

Comment:
Since this could also be a union table reference that refers to multiple tables, I think it's best to rely on the `applyAffectedTableNames` method.

```suggestion
		tableReference.applyAffectedTableNames( this::registerAffectedTable );
```